src/components/CompactDashboard/CompactDashboard.tsx

```diff
@@ -0,0 +1,197 @@
+
+const GUTTER_SIZE = 3;
+const CELL_WIDTH = 330;
+
+function sortFarmServers(ob1: { status: number, serverName: string }, ob2: { status: number, serverName: string }) {
+    if (ob1.status > ob2.status) {
+        return 1;
+    } else if (ob1.status < ob2.status) {
+        return -1;
+    }
+
+    if (ob1.serverName < ob2.serverName) {
+        return -1;
+    } else if (ob1.serverName > ob2.serverName) {
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+function sortFarms(ob1: { farmName: string }, ob2: { farmName: string }) {
+    if (ob1.farmName < ob2.farmName) {
+        return -1;
+    }
+
+    if (ob1.farmName > ob2.farmName) {
+        return 1;
+    }
+    return 0;
+}
+
+function checkFilter(filter: string, serverName: string) : boolean {
+    return serverName.toLowerCase().trim().indexOf(filter.toLowerCase().trim()) !== -1;
+}
+
+export class CompactDashboard extends React.Component<ICompactDashboardProps, any> {
+
+    collection: any;
+    list: any;
+
+    constructor(props?: ICompactDashboardProps) {
+        super(props);
+        this.state = { columnYMap: [], collection: undefined, list: undefined };
+    }
+
+    @autobind
+    private componentDidUpdate(prevProps: ICompactDashboardProps, prevState) {
+        if (this.props.isVertical === false && prevProps.isVertical === true) {
+            this.setState({ columnYMap: [] });
+        }
+        if (this.props.filter !== prevProps.filter) {
+            if (this.collection) {
+                this.setState({ columnYMap: [] }, () => {
+                    this.collection.recomputeCellSizesAndPositions();
+                });
+            }
+            if (this.list) {
+                this.list.recomputeRowHeights();
+            }            
+        }       
+    }
+
+
+    public render() { 
+        let {title, farms} = this.props;
+        let classname = classNames({ [this.props.className]: this.props.className !== undefined });
+        return (
+            <div className={classname}>
+                {
+                    this.props.isVertical &&
+                    <div className="compact-dashboard-container vertical">
+                        {
+                            <AutoSizer  >
+                                {({ width, height}) => (
+                                    <Collection
+                                        filt={this.props.filter}
+                                        ref={(reference) => {
+                                            this.collection = reference;
+                                        }}
+                                        cellCount={this.props.farms.length}
+                                        cellRenderer={this._renderRow}
+                                        cellSizeAndPositionGetter={function (index) {
+                                            return this.cellSizeAndPositionGetter(width, index);
+                                        }.bind(this)}

```

Comment:
this binds the function every time it renders. you can bind this in constructor or using autobind decorator